client/src/app/city-detail/city-detail.component.ts

```diff
@@ -25,10 +25,18 @@ export class CityDetailComponent implements OnInit {
     blue: 0,
   };
 
-  constructor(private auth: AuthService) {}
+  constructor(private auth: AuthService, private city: CityService) {}
 
   ngOnInit(): void {
+    // user data
     this.userId = this.auth.GetUserData().id;
+    // user resources and production
+    this.getAllData();
+    // refreash signal
+    this.city.refreashSignal.subscribe((data) => this.getAllData());
```

Comment:
Why not have a `UserService` or something that handles _all_ of user’s data? Store it in a `BehaviorSubject` that can be subscribed for changes and can hold its current value.

Then, all components that need to use any user data can just subscribe to it and update the displayed information whenever an update comes — the observable (here `BehaviorSubject`) handles the notifications about changes of the data.